tools/src/android-update-native-dependencies/updateChangelogFiles.ts

```diff
@@ -0,0 +1,36 @@
+import * as Changelogs from '../Changelogs';
+import dependenciesChangelogs from '../data/androidDependenciesChangelogs.json';
+import { AndroidProjectDependenciesUpdates } from './types';
+
+function maybeWrapInChangelogLink(text: string, dependencyName: string) {
+  const changelog = dependenciesChangelogs[dependencyName];
+  return changelog ? `[${text}](${changelog})` : text;
+}
+
+export async function addChangelogEntries(
+  updatesList: AndroidProjectDependenciesUpdates[]
+): Promise<void> {
+  await Promise.all(
+    updatesList
+      .filter((update) => update.updates.length > 0)
+      .filter((update) => update.report.changelogPath)
+      .map(async ({ updates, report }) => {
+        const changelog = Changelogs.loadFrom(report.changelogPath!);
+
+        await changelog.insertEntriesAsync(
+          Changelogs.UNPUBLISHED_VERSION_NAME,
+          Changelogs.ChangeType.LIBRARY_UPGRADES,
+          null,
+          updates.map(
+            (update) =>
+              `Updated Android native dependency: ${maybeWrapInChangelogLink(
+                `\`${update.fullName}:${update.oldVersion}\` ➡️ \`${update.newVersion}\``,
+                update.fullName
+              )}`
+          )
```

Comment:
I think this should be done outside the `insertEntriesAsync` invocation — if a single argument takes more than one line, it starts disrupting the reader to understand the flow.